apps/emqx_ds_builtin_raft/src/emqx_ds_replication_layer_meta.erl

```diff
@@ -684,40 +695,45 @@ node_sites(Node) ->
 
 ensure_tables() ->
     ok = mria:create_table(?META_TAB, [
-        {rlog_shard, ?SHARD},
+        {rlog_shard, ?RLOG_SHARD},
         {type, ordered_set},
         {storage, disc_copies},
         {record_name, ?META_TAB},
         {attributes, record_info(fields, ?META_TAB)}
     ]),
     ok = mria:create_table(?NODE_TAB, [
-        {rlog_shard, ?SHARD},
+        {rlog_shard, ?RLOG_SHARD},
         {type, ordered_set},
         {storage, disc_copies},
         {record_name, ?NODE_TAB},
         {attributes, record_info(fields, ?NODE_TAB)}
     ]),
     ok = mria:create_table(?SHARD_TAB, [
-        {rlog_shard, ?SHARD},
+        {rlog_shard, ?RLOG_SHARD},
         {type, ordered_set},
         {storage, disc_copies},
         {record_name, ?SHARD_TAB},
         {attributes, record_info(fields, ?SHARD_TAB)}
     ]),
     ok = mria:create_table(?TRANSITION_TAB, [
-        {rlog_shard, ?SHARD},
+        {rlog_shard, ?RLOG_SHARD},
         {type, bag},
         {storage, disc_copies},
         {record_name, ?TRANSITION_TAB},
         {attributes, record_info(fields, ?TRANSITION_TAB)}
     ]),
     ok = mria:wait_for_tables([?META_TAB, ?NODE_TAB, ?SHARD_TAB]).
```

Comment:
Q: don't we need to ensure the legacy tables so their content is loaded when we attempt the migration?

Reply:
I'm not sure, would be grateful to hear if it's indeed required from safety perspective. FWIW I haven't seen any troubles in my local tests due to the lack of `mria:wait_for_tables/1` on legacy tables. My current assumption is that if mria/mnesia has started, then the existing tables should be available to transactions. Besides, it's possible there are actually no legacy tables, so waiting for them will hang.

Reply:
Maybe they'll always need to be created then.  :thinking: 

IIRC, tables are loaded asynchronously, that's why there's `mnesia:wait_for_tables`.  If we require going over the legacy table contents to migrate, they _could_ be unavailable at that time if not waited on.

Quoting the `mnesia` docs:

> Mnesia startup is asynchronous. The function call mnesia:start() returns the atom ok and then starts to initialize the different tables. Depending on the size of the database, this can take some time, and the application programmer must wait for the tables that the application needs before they can be used. This is achieved by using the function mnesia:wait_for_tables/2.

I'd like to hear from someone with more experience in that are too.  :sweat_smile:

Reply:
RIght, thanks for the explanation. Hoping that c1e77f6c868cf51f061f935d58c99401b3639152 will take care of that.